src/server/adminInviteUser.ts

```diff
@@ -36,30 +47,31 @@ export async function adminInviteUser(
     });
 
     if (error) {
+        const logId = await logErrorReturnLogId(`Error with inviting user: ${error.message}`);
         return {
             data: null,
-            error: { "Failed to create user": error.message },
+            error: { type: "Invite User Failure", logId: logId },
         };
     }
 
-    if (data) {
-        const { error: createRoleError } = await supabase.from("profiles").insert({
-            primary_key: data.user.id,
-            role: userDetails.role,
-            first_name: userDetails.firstName,
-            last_name: userDetails.lastName,
-            telephone_number: userDetails.telephoneNumber,
-        });
-        if (createRoleError) {
-            return {
-                data: null,
-                error: { "Failed to create user profile": createRoleError.message },
-            };
-        }
-        void logInfoReturnLogId(
-            `Created a profile for ${userDetails.role} user: ${userDetails.email}`
+    const { error: createRoleError } = await supabase.from("profiles").insert({
+        role: userDetails.role,
+        first_name: userDetails.firstName,
+        last_name: userDetails.lastName,
+        telephone_number: userDetails.telephoneNumber,
+        user_id: data.user.id,
+    });
+
+    if (createRoleError) {
+        const logId = await logErrorReturnLogId(
+            `Error with insert profile: ${createRoleError.message}`
         );
```

Comment:
why not log the error object
```suggestion
        const logId = await logErrorReturnLogId(
            `Error with insert profile`,
            { error: createRoleError }
        );
```

Reply:
sure